crates/transpiler/src/passes/optimize_1q_gates_decomposition.rs

```
@@ -93,21 +93,23 @@ pub fn run_optimize_1q_gates_decomposition(
                     .iter()
                     .enumerate()
                     .filter_map(|(idx, gates)| {
-                        if !gates
-                            .iter()
-                            .all(|gate| basis_gates.as_ref().unwrap().contains(gate))
-                        {
+                        if !gates.iter().all(|gate| {
+                            basis_gates
+                                .expect("the target path always provides a hash set")
+                                .contains(gate)
+                        }) {
                             return None;
                         }
                         let basis = EULER_BASIS_NAMES[idx];
                         Some(basis)
                     })
                     .for_each(|basis| target_basis_set.add_basis(basis)),
                 None => match &global_decomposers {
-                    Some(bases) => bases
-                        .iter()
-                        .map(|basis| EulerBasis::__new__(basis).unwrap())
```

FWIW, at least in the original version of the port to rust the unwrap here was fine because we validate the global decomposers list Python side are all valid euler bases entries. I agree it's better to return the error now though since from C I don't think we do the same validation of the user input.

---

I couldn't prove it to myself from the codebase so I made it an exception (otherwise I'd have made it an `expect` with the reason) - mostly I'm just trying to remove `unwrap`, since we're now calling the functions from more than one place, and more places are passing direct user input to Rust space, so it's harder to be sure of the invariants.